src/packages/configuration-service.ts

```diff
@@ -34,141 +36,162 @@ interface IReverseClientAssetMapping {
     [assetId: string]: string;
 }
 
-interface IClientConfig {
+export interface IClientConfig {
     assetMapping: IClientAssetMapping;
     assetList: IGlobalAssetList;
     nameserviceConfiguration?: IBlockstackServiceInputOptions;
 }
 
-export interface IResolvedClientAssetMapping {
+export interface IResolvedClientAssetMap {
     [currencySymbol: string]: IGlobalAsset;
 }
 
-export abstract class NameServiceConfigurationService {
-    constructor() {
-        log.info(`Initizing NameServiceConfigurationService with options:- `);
-    }
-
-    public abstract getResolvedClientAssetMapping = async (): Promise<IResolvedClientAssetMapping> => ({});
-    public abstract translateSymbolToAssetId = async (currencySymbol: string): Promise<string> => ("");
-    public abstract translateAssetIdToSymbol = async (assetId: string): Promise<string> => ("");
-}
-
-export class BlockstackConfigurationService extends NameServiceConfigurationService {
+export class ConfigurationService {
 
-    private reverseClientAssetMapping?: IReverseClientAssetMapping;
-    private blockstackNameservice: nameservice.BlockstackService;
-    private clientName: string;
-    private clientConfig?: IClientConfig;
-    private clientAssetMapping?: IClientAssetMapping;
-    private blockstackID: string | undefined;
-    private globalAssetMap?: IGlobalMap;
+    public clientName: string;
+    public clientConfig?: IClientConfig;
+    public globalAssetMap?: IGlobalMap;
+    public clientAssetMapping?: IClientAssetMapping;
+    public reverseClientAssetMapping?: IReverseClientAssetMapping;
+    public resolvedClientAssetMap?: IResolvedClientAssetMap;
+    public nameServiceConfig?: IBlockstackServiceInputOptions;
 
-    constructor(clientName: string, cruxID?: string) {
-        super();
+    constructor(clientName: string) {
         this.clientName = clientName;
-        if (cruxID) {
-            this.blockstackID = identityUtils.IdTranslator.cruxToBlockstack(identityUtils.CruxId.fromString(cruxID)).toString();
-        }
-        this.blockstackNameservice = new nameservice.BlockstackService({ domain: this.clientName + identityUtils.CRUX_DOMAIN_SUFFIX });
         log.info(`BlockstackConfigurationService initialised with default configs`);
     }
 
     public init = async () => {
-        this.clientConfig = await this._getClientConfig();
-        this.globalAssetMap = await this._getGlobalMapping();
-        this.clientAssetMapping = await this._getClientAssetMapping();
-        this.reverseClientAssetMapping = await this._getReverseClientAssetMapping();
+        await this._setupClientConfig()
+            .then(() => this._setupGlobalAseetMap())
+            .then(() => this._setupClientAssetMapping())
+            .then(() => this._setupReverseClientAssetMapping())
+            .then(() => this._setupResolvedClientAssetMapping())
+            .then(() => this._setupNameServiceConfig());
```

Comment:
Is there any scope to make few calls parallel, if yes then let's execute them in parallel.